src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -608,6 +608,22 @@ namespace xclcpuemhal2 {
     sock = new unix_socket;
   }
 
+  void CpuemShim::getCuRangeIdx() {
+    std::string instance_name = "";
+    for (const auto& kernel : m_xclbin.get_kernels()) {
+      // get properties of each kernel object
+      const auto& props = xrt_core::xclbin_int::get_properties(kernel);
+      //get CU's of each kernel object.iterate over CU's to get arguments
+      if (props.address_range != 0 && !props.name.empty())
+        continue;       
+      for (const auto& cu : kernel.get_cus()) {
+        instance_name = cu.get_name();
+        if (!instance_name.empty())
+          mCURangeMap[instance_name] = props.address_range;
```

Comment:
```suggestion
          mCURangeMap.emplace(std::move(instance_name), props.address_range);
```
This assumes that the instance_name element did not already exist, which seems to be the case.